be/src/exec/tablet_sink.cpp

```diff
@@ -1095,21 +1123,106 @@ bool OlapTableSink::is_full() {
         for_each_index_channel([&full](NodeChannel* ch) { full |= ch->is_full(); });
     }
 
-    return full;
+    return full || _is_automatic_partition_running.load(std::memory_order_acquire);
+}
+
+Status OlapTableSink::_automatic_create_partition() {
+    TCreatePartitionRequest request;
+    TCreatePartitionResult result;
+    request.__set_txn_id(_txn_id);
+    request.__set_db_id(_vectorized_partition->db_id());
+    request.__set_table_id(_vectorized_partition->table_id());
+    request.__set_partition_values(_partition_not_exist_row_values);
+    TNetworkAddress master_addr = get_master_address();
+    RETURN_IF_ERROR(ThriftRpcHelper::rpc<FrontendServiceClient>(
+            master_addr.hostname, master_addr.port,
+            [&request, &result](FrontendServiceConnection& client) { client->createPartition(result, request); }));
```

Comment:
If the master change during this period, it will failed.

Reply:
rpc may be failed by many reason, it doesn't matter.